### PR TITLE
PLAT-502 Add Redis support for v2

### PR DESCRIPTION
**Pull request by katekligman**
## Description
This pull request adds Redis support for version 2 of django-segments.  Long live django-segments!

## Changelog
* SegmentMemberships have been completely replaced with a Redis backend to improve scaling capacity.
* Segments have priority and sort in descending order.  Priority is editable on a per segment basis in the django admin.
* Running sets of users with recently updated segments are stored in Redis and kept for one week.
* Inline segment memberships have been removed from the django admin user view.

### CODE REVIEW

File: segments/admin.py [hunk 1]
```diff
@@ -11,13 +11,10 @@ class SegmentAdmin(admin.ModelAdmin):
...
-    list_display = ('name', 'members_count', 'definition', 'manager_method')
-    readonly_fields = ('created_date', 'members_count', 'static_users_sample', 'sql_users_sample')
-    fields = ('name', 'slug', 'members_count', 'definition', 'manager_name', 'manager_method',
-              'sql_users_sample', 'static_ids', 'static_users_sample', 'created_date')
-
-    def members_count(self, segment):
-        return len(segment)
+    list_display = ('name', 'priority', 'members_count', 'definition')
+    readonly_fields = ('created_date', 'members_count', 'updated_date', 'recalculated_date')
+    fields = ('name', 'slug', 'priority', 'members_count', 'definition', 'created_date', 'updated_date', 'recalculated_date')
```
Comment (Anton-Shutik) [inline segments/admin.py]: Why do we need the `fields` here ? Django will add it itself

File: segments/models.py [hunk 2]
```diff
@@ -302,29 +135,8 @@ class SegmentMixin(object):
     @property
     def segments(self):
         """Return all the segments to which this member belongs."""
-        return Segment.objects.filter(id__in=self.segment_set.all().values_list('segment_id', flat=True))
+        return Segment.objects.filter(id__in=SegmentHelper().get_user_segments(self.pk)).order_by('-priority')
```
Comment (Anton-Shutik) [inline segments/models.py]:
```suggestion
        return Segment.objects.filter(id__in=Segment.helper.get_user_segments(self.pk)).order_by('-priority')
```

File: segments/router.py [hunk 3]
```diff
@@ -20,19 +20,5 @@ def db_for_read(self, model, **hints):
...
-        Namely this targets Segment.refresh() where something like this happens:
-
-            SegmentMembership(segment=s, user=u)
-
-        Where 'u' is coming from a SEGMENTS_EXEC_CONNECTION queryset.
-        """
-        is_segments = obj1._meta.app_label == 'segments'
-        is_conn = obj2._state.db == SEGMENTS_EXEC_CONNECTION
-        return True if is_segments and is_conn else None
-
```
Comment (Anton-Shutik) [inline segments/router.py]: Do we need the router at all ? Looks like it does not have any logic, I think we can remove it

File: segments/models.py (hunk 2 above)
Comment (katekligman) [reply segments/models.py]: 👍

File: segments/admin.py (hunk 1 above)
Comment (katekligman) [reply segments/admin.py]: I'm using the fields to define the order.

**Pull request merged**